packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -558,21 +558,103 @@ abstract class ModalRoute<T> extends TransitionRoute<T> with LocalHistoryRoute<T
+  ///     );
+  ///   },
+  /// );
+  ///```
+  ///
+  /// We've used PageRouteBuilder to demonstrate the buildTransitions method
+  /// here. The body of an override of the buildTransitions method would be
+  /// defined in the same way.
+  ///
+  /// When the Navigator pushes a route on the top of its stack, the
+  /// secondaryAnimation can be used to define how route that was on the top
+  /// of the stack leaves the screen. Similarly when the topmost route is
+  /// popped, the secondaryAnimation can be used to define how the route
+  /// below it reappears on the screen. When the Navigator pushes a new route
+  /// on the top of its stack, the old topmost route's secondaryAnimation
+  /// runs from 1.0 to 0.0.  When the Navigator pops the topmost route, the
+  /// secondaryAnimation for the route below it runs from 0.0 to 1.0.
+  ///
+  /// The example below adds an transition that's driven by the
+  /// secondaryAnimation. When this route disappears because a new route has
+  /// been pushed on top of it, it translates in the opposite direction of
+  /// the new route. Likewise when the route is exposed because the topmost
+  /// route has been popped off.
+  ///
+  /// ```dart
+  ///   transitionsBuilder: (
+  ///       BuildContext context,
+  ///       Animation<double> primaryAnimation,
+  ///       Animation<double> secondaryAnimation,
+  ///       Widget child) {
+  ///     return new SlideTransition(
+  ///       position: new FractionalOffsetTween(
+  ///         begin: FractionalOffset.bottomLeft,
+  ///         end: FractionalOffset.topLeft,
+  ///       ).animate(primaryAnimation),
+  ///       child: new SlideTransition(
+  ///         position: new FractionalOffsetTween(
+  ///           begin: FractionalOffset.topLeft,
+  ///           end: FractionalOffset.bottomLeft,
+  ///         ).animate(secondaryAnimation),
+  ///         child: child,
+  ///       ),
+  ///     );
+  ///   }
+  ///```
+  ///
+  /// In practice the secondaryAnimation is not used very often because animating
+  /// both the topmost route and the route below it can be more distracting
+  /// than helpful.
+  ///
   /// * [context] The context in which the route is being built.
-  /// * [animation] The animation for this route's transition. When entering,
+  /// * [primaryAnimation] The animation for this route's transition. When entering,
   ///   the animation runs forward from 0.0 to 1.0. When exiting, this animation
   ///   runs backwards from 1.0 to 0.0.
-  /// * [forwardAnimation] The animation for the route being pushed on top of
-  ///   this route. This animation lets this route coordinate with the entrance
-  ///   and exit transition of routes pushed on top of this route.
-  Widget buildTransitions(BuildContext context, Animation<double> animation, Animation<double> forwardAnimation, Widget child) {
+  /// * [secondaryAnimation] The animation for the route below the topmost route.
```

Comment:
This whole paragraph sounds like it's referring to something that's not 'self' but the API is still asking the user to build animated widgets associated with the same 'self' route. 

Perhaps 'The animation for this route when it's the second topmost route on the stack. It runs from 0.0 to 1.0 as another route is pushed on top of this one and from 1.0 to 0.0 as another route above this one is popped. 

This animation runs together with the [primaryAnimation] of the route above it'?